cmd/vcert/validators.go

```diff
@@ -58,9 +59,15 @@ func validateCommonFlags(commandName string) error {
 		return fmt.Errorf("The '-file' option cannot be used used with any other -*-file flags. Either all data goes into one file or individual files must be specified using the appropriate flags")
 	}
 
-	csrOptionRegex := regexp.MustCompile(`^file:.*$|^local$|^service$|^$`)
+	csrOptionRegex := regexp.MustCompile(`(^file:).*$|^local$|^service$|^$`)
 	if !csrOptionRegex.MatchString(flags.csrOption) {
-		return fmt.Errorf("unexpected -csr option: %s", flags.csrOption)
+		return fmt.Errorf("unexpected -csr option provided: %s; specify one of the following options: %s, %s, or %s", flags.csrOption, "'file:<filename>'", "'local'", "'service'")
+	}
+
+	csrOptFlagResults := csrOptionRegex.FindStringSubmatch(flags.csrOption)
+
+	if csrOptFlagResults[1] != "" && (flags.keyTypeString != "" || flags.keyCurveString != "" || flags.keySize > 0) {
+		return fmt.Errorf("the '-keytype','-keycurve' and 'key-size' options cannot be used when '-csr file:' option is provided")
```

Comment:
Should be `--key-type`, `--key-curve`, `--key-size`, and `--csr`.

Reply:
@tr1ck3r Updated the error messages with the -- as requested.